sys/net/bpf_zerocopy.c

```diff
@@ -216,7 +216,7 @@ zbuf_setup(struct thread *td, vm_offset_t uaddr, size_t len,
 	map = &td->td_proc->p_vmspace->vm_map;
 	for (i = 0; i < zb->zb_numpages; i++) {
 		zb->zb_pages[i] = zbuf_sfbuf_get(map,
-		    uaddr + (i * PAGE_SIZE));
+		    (char * __capability)uaddr + (i * PAGE_SIZE));
```

Comment:
Does this cast really work?

Reply:
I've asked myself the same question several times, probably for the same reason. It turns out that diff is stupid and displays the OLD version of the function signature when displaying context of a function with an altered signature... `uaddr` is actually `void * __capability uaddr` so the cast is fine.

Reply:
Ohhhhhhhhhh wow that's confusing though displaying the new version of the function signature could also get just as confusing in certain cases, but probably more helpful overall...